fem/qspace.hpp

```diff
@@ -94,6 +101,15 @@ class QuadratureSpaceBase
    /// Write the QuadratureSpace to the stream @a out.
    virtual void Save(std::ostream &out) const = 0;
 
+   /// Return the integration weights (including geometric factors).
+   const Vector &GetWeights() const;
+
+   /// Return the integral of the scalar Coefficient @a coeff.
+   double Integrate(Coefficient &coeff) const;
+
+   /// Return the integral of the VectorCoefficient @a coeff in @a integrals.
+   void Integrate(VectorCoefficient &coeff, Vector &integrals) const;
```

Comment:
Might be a note for future updates but these methods and the `QuadratureFunction::Integrate(Vector...)` one create a number of potentially large temporary memory objects. It might be useful to utilize the work in https://github.com/mfem/mfem/pull/4065 here.

Reply:
Yes, definitely. We should make note of this, so then when something like #4065 is merged, we take advantage of it here. (This is exactly the intended use-case).